### DISCUSSION TITLE
Adding servers to Modrinth

### DISCUSSION THREAD

**Discussion by dgxo** [modrinth/.github#35] [Feature requests]
### Is your feature request related to a problem? Please describe.

You often see server lists that are filled with servers that pay to be at the top with flashy, attention seeking banners, almost always only seeking monetary profit and unlikely to have a very good player experience. It's usually difficult to find a good server that provides a good high quality experience without paywalls or limits. I think having a server list built into Modrinth would allow small creators and teams that aren't trying to extract money from every bone in their players' body to be able to be discovered and enjoyed by the Minecraft playerbase.

### Describe the solution you'd like

Similar to how shaders and datapacks were recently added, I believe that having a Servers tab much like Minecraft MP, MC Server List and other websites could help drive more traffic towards Modrinth and open more possibilities for creators to make use of Modrinth's superior features and management, as opposed to the current abundance of detached, biased and unfair server lists, that make it difficult for casual and professional players to find high quality and reliable servers, without being drowned out by huge promoted and corrupted networks.

### Describe alternatives you've considered

Although it may be more difficult to tie in this system into Modrinth's current backend than things like shaders or datapacks, it would be easier for users to access this section of Modrinth than something like a separate website or service. This seems like pretty much the only option to add this feature without making things more complex for players interacting with the site.

### Additional context

I've seen a bit of discussion about this in the Discord, and a few mentions about the large networks hogging all the attention in existing server lists, so I think the general Minecraft community would enjoy this feature, and especially small server owners, bringing them above the almost monopolies of servers like Purple Prison and Complex Gaming.

If you want to see this, maybe add a thumbs under this up so the devs can see the community's reaction.

**Comment by ghost** [comment]
This would need a lot of restructuring to how a project is presented to a user, a server is not something you download, so versions should be replaced with announcements, there should be a more prominent field with the server's IP, and all cards on the page are to be moved around. Compex changes to the way a kind of project is stored and presented would probably require to wait until api v3 to even think of implementing a server list.

**Reply by dgxo** [reply]
I agree, although I think Modrinth may not be the place to host announcements for servers as this isn't a feature on other types of submissions like mods and datapacks. It would probably be simpler for server owners to have updates posted on a Discord or similar, maybe forum or blog, removing the need for this on a separate website, that the community might find more difficult to access or find it unfamiliar. So instead of replacing and redesigning the downloading and version UI, it could just be removed or inaccessible. This would be easier to implement as it means removing a part of the front-end and API, instead of integrating a new feature (plus it could be introduced into the entire website at some point, if that's where Modrinth is planning on going).
I do think some changes are definitely needed in order to support servers, but they could be released possibly in a few weeks as an experimental feature with just the versions section removed, letting owners display the server IP themselves and slowly introducing more features to fully reinforce this idea and let servers move away from the current mess of server lists.

**Reply by ghost** [reply]
Modrinth updates would be cool, as an optional feature for server owners to opt into. You could just use the same follow feature to get updates on how a server's doing.

It would also be cool to have a section where you can connect a modpack project that you reccomend/require using on your server.

**Reply by MagmaSlime123** [reply]
I'd love this tbh. Server lists are messy, P2W-y, and lots of other not-so-good things.

**Reply by Lancelotbronner** [reply]
Why try to present them as projects in the backend? They’re not.

just adding a separate route /servers instead of trying to hack into /projects would be much better

**Reply by ghost** [reply]
That's the issue, you really can't have much freedom with the current backend. That would need to wait until api v3 or v4

**Reply by Lancelotbronner** [reply]
I mean, additions aren't breaking changes and I'm suggesting to make projects and servers two completely separate entities so I don't see why it couldn't be done as part of v2

Then again I don't play public servers so I don't really care when or if it happens

I just think trying to make projects flexible enough to support things that aren't projects is a bad idea

**Reply by triphora** [reply]
@HyperSoop It is possible for us to add stuff separately without following the same "project" structure.

**Reply by ghost** [reply]
Nice
Then it just depends on how easy it would be to add
